truffle/src/main/ruby/core/complex.rb

```diff
@@ -53,9 +53,10 @@ class Complex < Numeric
   undef_method :i
 
   def self.convert(real, imag = undefined)
-    if real.equal?(nil) || imag.equal?(nil)
+    if nil.equal?(real) || nil.equal?(imag)
```

Comment:
Yeah, there are a few places in core where it's not like this.
For `undefined` you can't forget as it will trigger an assert if you write it in the other way :)